### PR TITLE
Add book category logic and some package refactor:

### PR DESCRIPTION
**Pull request by DEz-1996**
1) add category entity, controller, service, repository and Dtos
2) change book entity, controller, service, repository and Dtos with relation to book category
3) refactor dtos to corresponding folders

### CODE REVIEW

**Review by d1sam — CHANGES_REQUESTED** [review]
why do I see changes related to **all** spring boot hws?

File: src/main/java/online/book/store/controller/CategoryController.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    @GetMapping("/{id}")
+    @Operation(summary = "Get category by ID", description = "Get the existing category by ID")
+    public CategoryDto getCategoryById(@PathVariable Long id) {
+        return categoryService.getById(id);
+    }
+
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PostMapping
+    @Operation(summary = "Create a new category", description = "Create a new category")
+    public CategoryDto createCategory(
```
Comment (vovasalyha) [inline src/main/java/online/book/store/controller/CategoryController.java:55]:
```suggestion
    @ResponseStatus(HttpStatus.CREATED)
    public CategoryDto createCategory(
```

File: src/main/java/online/book/store/model/Book.java [hunk 2]
```diff
@@ -32,4 +36,9 @@ public class Book {
     private String coverImage;
     @Column(nullable = false)
     private boolean isDeleted = false;
+    @ManyToMany
+    @JoinTable(name = "books_categories",
+            joinColumns = @JoinColumn(name = "book_id"),
+            inverseJoinColumns = @JoinColumn(name = "category_id"))
+    private Set<Category> categories;
```
Comment (vovasalyha) [inline src/main/java/online/book/store/model/Book.java:47]: Ignore it in equals, hashCode and toString.

File: src/main/java/online/book/store/service/imp/CategoryServiceImp.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        Category category = categoryMapper.toEntity(categoryRequestDto);
+        Category savedCategory = categoryRepository.save(category);
+        return categoryMapper.toDto(savedCategory);
+    }
+
+    @Override
+    public CategoryDto update(CreateCategoryRequestDto categoryRequestDto, Long id) {
+        Category category = categoryMapper.toEntity(categoryRequestDto);
+        category.setId(id);
+        boolean isIdPresent = categoryRepository.findById(id).isPresent();
```
Comment (vovasalyha) [inline src/main/java/online/book/store/service/imp/CategoryServiceImp.java]: Better use existsById method.

**Review by vovasalyha — APPROVED** [review]
Approving, but see few minor comments.

**Pull request merged**